app/src/main/java/io/halkyon/services/BindApplicationService.java

```diff
@@ -66,13 +73,18 @@ private void unMountSecretVolumeEnvInApplication(Claim claim) throws ClusterConn
 
     private void createSecretForApplication(Claim claim, Credential credential, String url)
             throws ClusterConnectException {
+        Map<String, String> vaultSecret = kvSecretEngine.readSecret(credential.vaultKvPath);
```

Comment:
Can you add a if/then block to generate an error, return a proper message when we cannot find for the Path searched the credential ?